### ISSUE TITLE
Massively repeated characters on MacOS

### ISSUE THREAD

**Issue by exvayn** [argosopentech/argos-translate#197]
```
argos-translate --from-lang en --to-lang ru "Hello World"
>>>  З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З! З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З! З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З
```

or another example

```
argos-translate --from-lang en --to-lang ru "Good to see you"
>>>  У! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До! До!
```

tried script
agro-translate.py:
```
import argostranslate
from argostranslate import package, translate
package.install_from_path('translate_en_ru.argosmodel')

installed_languages = translate.get_installed_languages()
for lang in installed_languages:
  print(lang)
 
translation_en_ru = installed_languages[0].get_translation(installed_languages[1])
translated = translation_en_ru.translate("How are you!")
print(translated)
```
and got:
```
python agro-translate.py 
>>> English
>>> Russian
>>> ?!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

what did i do wrong?
Mac OS 11.1 Big Sur

**Comment by PJ-Finlay** [comment]
That is strange, it's possible that the neural net just has an unexpected output but I can't reproduce these examples.

Can you try running with:
```
export DEBUG=1

```

**Comment by farfromrefug** [comment]
@DanijelCvit i have the same issue and sadly clearing the cache did not fix it. I am using `argos-translate-gui`  1.5.2. Anything else i can do?

**Comment by PJ-Finlay** [comment]
What platform are you on?

**Comment by farfromrefug** [comment]
@PJ-Finlay sorry i am on macos 11.6 and installed through libretranslate with pip

**Comment by PJ-Finlay** [comment]
This looks like an unknown issue with inference on Mac.

**Comment by farfromrefug** [comment]
@PJ-Finlay i should mention that it had been working for months. It broke once i updated libretranslate(not sure what version i was on). Now i cant get it to work. I mostly use fr/en but tested also with es and it does the same

**Comment by PJ-Finlay** [comment]
That's interesting z it could be caused by upgrading CTranslate2.

**Comment by guillaumekln** [comment]
Since someone got it working after clearing the cache, it could be a corrupted model package. In CTranslate2 2.6 we added better error checking when loading models (before that some errors were silent). @PJ-Finlay Is this minimum version included in any release of Argos Translate?

> It broke once i updated libretranslate(not sure what version i was on)

@farfromrefug It would definitely help if you can find when it broke. You could iteratively downgrade the package to find the first version causing the issue.

Also, are we talking M1 Mac or Intel Mac here?

**Comment by PJ-Finlay** [comment]
CTranslate2 2.6 should be available in argostranslate>=1.5.3.

I think clearing the cache may have fixed a different issue but I'm not sure.

I'm also curious if the issue happens for all models/languages or just Spanish and French.

**Comment by farfromrefug** [comment]
@guillaumekln i ll try and run some tests and rollback
@PJ-Finlay when you say standalone mac app you are talking about `argos-translate-gui` right?

**Comment by farfromrefug** [comment]
@guillaumekln @PJ-Finlay i was a bit tough because of icu but i managed to rollback to:
* libretranslate 1.2.3
* argostranslate 1.1.4
* ctranslate2 1.17.1

I did not have to redownload language packages (i have the ones from 1.5.3) and it works again.
Now before that i tried to upgrade argostranslate to 1.5.3 and CTranslate2 but it did not fix the issue.

**Comment by PJ-Finlay** [comment]
There's a Mac .app file release on GitHub packaged for Mac that runs an older version of Argos Translate. Installing from PyPI or source also has `argos-translate-gui` but the latest version.

**Comment by farfromrefug** [comment]
@PJ-Finlay ok i see that s what i was expecting. Now there is still an issue somewhere which seems to have started in 1.2.4 or 1.2.5. Hope we can fix it

**Comment by PJ-Finlay** [comment]
LibreTranslate 1.2.3 (before the issue occurs) uses Argos Translate 1.1.4 which uses CTranslate2 1.17.1.

**Comment by guillaumekln** [comment]
I was able to reproduce the issue on an old MacBook Air. The issue was introduced in CTranslate2 2.4.0 by this change:

> Compile the CPU INT8 quantization kernel with FMA instructions

The current workaround is to enable AVX instead of AVX2 for CTranslate2 optimized kernels:

```
% argos-translate --from-lang en --to-lang ru "Hello World"
З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З! З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З! З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З З

% CT2_FORCE_CPU_ISA=AVX argos-translate --from-lang en --to-lang ru "Hello World"
Привет мир
```

I will report back to this issue when a definitive fix is released. Thank you for reporting.

**Comment by guillaumekln** [comment]
This issue should be fixed with CTranslate2 2.10.1.

Also, it seems the workaround shown above only worked for this specific example and not for the general case, so updating to the latest version is recommended.
